src/controllers/user-controller.js

```diff
@@ -210,22 +212,43 @@ router.get("/:id/grade", authenticateJWT, authorizeUser, async (req, res) => {
  */
 router.post("/:id/programs", authenticateJWT, authorizeUser, async (req, res) => {
     try {
-        const prospective = await Users.findById(req.params.id);
+        const prospective = await prospectiveStudent.findById(req.params.id);
         const programIds = req.body.programs;
 
         const programs = await Program.find({ _id: { $in:programIds }})
-
+        
         if (!prospective) {
             return res.status(404).json({ error: "존재하지 않는 학생입니다." })
         }
```

Comment:
## Problematic
Returning an error message saying that the student does not exist can be problematic.
As the domain of search is just the prospective students, if an existing user (perhaps a student), this message will be confusing.

Reply:
Will be solved later in the revision.